Add Storage API

jtjk opened the pull request:
Storage API included

jtjk committed:
Add Storage API

jmspring commented on azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/AzureComputeApi.java:
```diff
@@ -138,10 +138,11 @@ VirtualMachineApi getVirtualMachineApiForDeploymentInService(@PathParam("deploym
    /**
     * The Service Management API includes operations for managing the storage accounts in your subscription.
```
Should be "Resource Management API"

jtjk replied:
I made new commit and replaced that text to "Resource Management API"

jtjk commented:
Updated comment field in AzureComputeAPI.java

ritazh commented:
Running `StorageAccountApiLiveTest` getting the error below. I do see a new storage account has been created.

```
java.lang.AssertionError: expected [true] but found [false]
org.jclouds.azurecomputearm.features.StorageAccountApiLiveTest.testList(StorageAccountApiLiveTest.java:52)
```

Command used:

```
mvn -Dtest=StorageAccountApiLiveTest -Dtest.azurecompute-arm.identity="" -Dtest.azurecompute-arm.subscriptionid="" -Dtest.azurecompute-arm.credential="" -Dtest.azurecompute-arm.endpoint="https://management.azure.com" -Dtest.jclouds.oauth.resource="https://management.azure.com/" -Dtest.oauth.endpoint="https://login.microsoftonline.com/[TenantId]/oauth2/token" test
```

ritazh commented on azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/StorageAccountApiLiveTest.java:
```diff
@@ -0,0 +1,125 @@
+      assertNotNull(storage.storageServiceProperties().creationTime());
+   }
+
+   @Test()
```
You probably need `@Test(dependsOnMethods = "testCreate")` to ensure testCreate runs before testList so that the storage list is not empty. This seems to resolve the failed live test for me.

jmspring replied:
What I am wondering -
Without Rita's addition, the mvn test run lists (for me) 8 tests run, 2 fail.
With the addition, the man test run lists (for me) 7 tests run, 0 failed.

Why the difference in total test count?  Unless the depends takes the testCreate out of the tests run.  Rather than depend on create, there should be a setup and teardown method for testList.

jtjk commented:
I added setup and tearDown to take care that there is new resource group and storage for tests. Also I added new list method with tests to get storages only inside resource group and not all in subscription.

ritazh commented on azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/StorageAccountApi.java:
```diff
@@ -16,125 +16,146 @@
  * @see <a href="http://msdn.microsoft.com/en-us/library/azure/ee460790">docs</a>
  */
-@Path("/services/storageservices")
 @Headers(keys = "x-ms-version", values = "{jclouds.api-version}")
```
Forgot to remove this one

jtjk replied:
This is also changed now. I propose that for rest of these we create own ticket where we create one cleaning commit where we change all of these and also remove some obsolete files etc. I go and create ticket to trello

ritazh commented:
LGTM - Added a card to remove subscriptionid, tenantid in mock tests and mock json files.
